test/intl402/DateTimeFormat/prototype/formatRangeToParts/date-undefined-throws.js

```diff
@@ -0,0 +1,31 @@
+// Copyright 2019 Google, Inc.  All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description:
+  Throws a RangeError if startDate or endDate are undefined.
+info:
+  Intl.DateTimeFormat.prototype.formatRangeToParts ( startDate , endDate )
+
+  1. Let dtf be this value.
+  2. If Type(dtf) is not Object, throw a TypeError exception.
+  3. If dtf does not have an [[InitializedDateTimeFormat]] internal slot, throw a TypeError exception.
+  4. If startDate is undefined or endDate is undefined, throw a RangeError exception.
+
+features: [Intl.DateTimeFormat-formatRange]
+---*/
+var dtf = new Intl.DateTimeFormat();
+
+var date = new Date();
+
+assert.throws(RangeError, function() {
+  dtf.formatRangeToParts(undefined, undefined);
```

Comment:
Add tests for omitted arguments like `formatRangeToParts()`, `formatRangeToParts(date)`, please. (Also `formatRange`.)

Reply:
Agreed with suggestion. Plus, considering: 

```
4. If startDate is undefined or endDate is undefined, throw a RangeError exception.
5. Let x be ? ToNumber(startDate).
6. Let y be ? ToNumber(endDate).
```

This might be a good option to use a poisoned value instead of date. This way we can assert the RangeError always happens before evaluating the arguments any further:

```js
var poison = { valueOf() { throw new Test262Error(); } };

assert.throws(RangeError, function() {
  dtf.formatRangeToParts(undefined, poison);
});

assert.throws(RangeError, function() {
  dtf.formatRangeToParts(poison, undefined);
});

assert.throws(RangeError, function() {
  dtf.formatRangeToParts(poison);
});

assert.throws(RangeError, function() {
  dtf.formatRangeToParts(); // Not possible to poison this one
});
```

Reply:
this applies to .formatRange as well

Reply:
done